tests/boards/intel_adsp/smoke/src/cpus.c

```diff
@@ -120,7 +120,7 @@ static void core_smoke(void *arg)
 	       clk_ratios[cpu] / 1000, clk_ratios[cpu] % 1000);
 
 	for (int i = 0; i < cpu; i++) {
-		int32_t diff = MAX(1, abs(clk_ratios[i] - clk_ratios[cpu]));
+		int32_t diff = MAX(1, (uint32_t)llabs((int64_t)clk_ratios[i] - clk_ratios[cpu]));
```

Comment:
would a simpler `abs((int32_t)(clk_ratios[i] - clk_ratios[cpu]))` be enough?

Reply:
Clock is not fast enough to need the signed bit. So we can do that to avoid 64-bit math.